aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -1084,7 +1084,8 @@ SparseTensor& _sspaddmm_out_cpu(
       "sspaddmm: Argument #1: Expected dim 1 size ", dim_k, ", got ", t.size(1));
 
   int64_t nnz        = sparse._nnz();
-  LongTensor indices = sparse._indices();
+  // We have to make indices contiguous as we use indices.data_ptr in _to_csr which assumes row-contiguous storage  
```

Comment:
is it checked anywhere that inputs are coalesced? Does it work for uncoalesced inputs?

Reply:
sparse tensor is explicitly coalesced above:
https://github.com/pytorch/pytorch/blob/c92b005d99f3efa7c8c46cdde703bc022d751aa7/aten/src/ATen/native/sparse/SparseTensorMath.cpp#L1068

Normally, `test_shape` performs tests for coalesced and uncoalesced sparse tensors, `self._gen_sparse` generates those tensors using `self.genSparseTensor(with_size, sparse_dim, nnz, self.is_uncoalesced, self.device)`
https://github.com/pytorch/pytorch/blob/c92b005d99f3efa7c8c46cdde703bc022d751aa7/test/test_sparse.py#L1084-L1088